rust/frontend/src/function.rs

```diff
@@ -158,146 +150,88 @@ impl<'a, 'm> Builder<'a, 'm> {
     {
-        for arg in args {
+        args.into_iter().for_each(|arg| {
             self.arg(&arg);
-        }
+        });
         self
     }
 
     /// Sets an output for a function that requirs a mutable output to be provided.
     /// See the `basics` in tests for an example.
-    pub fn set_output<'b, T: 'b + ?Sized>(&mut self, arg: &'b mut T) -> Result<&mut Self>
+    pub fn set_output<T>(&mut self, ret: T) -> &mut Self
     where
-        TVMValue: From<&'b T>,
-        TVMTypeCode: From<&'b T>,
+        TVMRetValue: From<T>,
     {
-        if self.ret_buf.is_none() {
-            let tvm_ret =
-                unsafe { TVMRetValue::from_tvm_value(TVMValue::from(arg), TVMTypeCode::from(arg)) };
-            self.ret_buf = Some(tvm_ret);
-        } else {
-            bail!(ErrorKind::AtMostOneReturn)
-        }
-        Ok(self)
+        self.ret_buf = Some(ret.into());
+        self
     }
 
     /// Calls the function that created from `Builder`.
-    pub fn invoke(&mut self) -> Result<TVMRetValue> {
-        self.clone()(())
-    }
-}
-
-impl<'a, 'm> FnOnce<((),)> for Builder<'a, 'm> {
-    type Output = Result<TVMRetValue>;
-    extern "rust-call" fn call_once(self, _: ((),)) -> Self::Output {
-        if self.func.is_none() {
-            bail!("{}", ErrorKind::FunctionNotFound);
-        }
-
-        let mut ret_val = unsafe { mem::uninitialized::<ts::TVMValue>() };
-        let mut ret_type_code = 0 as c_int;
-        if self.arg_buf.is_some() {
-            let arg_buf = self.arg_buf?;
-            let mut num_args = arg_buf.len();
-            let mut values = arg_buf
-                .iter()
-                .map(|tav| tav.value.inner)
-                .collect::<Vec<ts::TVMValue>>();
-            let mut tcodes = arg_buf
-                .iter()
-                .map(|tav| tav.type_code as c_int)
-                .collect::<Vec<_>>();
-
-            if self.ret_buf.is_some() {
-                num_args = num_args + 1;
-                let ret_buf = self.ret_buf?;
-                let (ret_val, ret_type_code) = TVMRetValue::into_tvm_value(ret_buf);
-                values.append(&mut vec![ret_val.inner]);
-                tcodes.append(&mut vec![ret_type_code as c_int]);
-            }
-
-            values.truncate(num_args);
-            tcodes.truncate(num_args);
-            check_call!(ts::TVMFuncCall(
-                self.func?.handle,
-                values.as_mut_ptr(),
-                tcodes.as_mut_ptr(),
-                num_args as c_int,
-                &mut ret_val as *mut _,
-                &mut ret_type_code as *mut _
-            ));
-        } else {
-            check_call!(ts::TVMFuncCall(
-                self.func?.handle,
-                ptr::null_mut(),
-                ptr::null_mut(),
-                0 as c_int,
-                &mut ret_val as *mut _,
-                &mut ret_type_code as *mut _
-            ));
-        }
+    pub fn invoke(&mut self) -> Result<TVMRetValue, Error> {
```

Comment:
Since you removed `FnOnce` impl, is there any other place needing `unboxed_closure` feature? if none, please remove it from frontend.